### PR TITLE
Event creation

### PR DESCRIPTION
**Pull request by Atmos4**
Massive PR, let me know if I need to split it.
- A lot of visual improvements
- Simple validator class with 
  - decorator pattern validation creation, zod-style
  - field rendering
  - error display
- Pages to:
  - create an event
  - delete an event (still WIP)
  - publish/unpublish an event
  - edit an event

### CODE REVIEW

File: utils/form_validation.php [hunk 1]
```diff
@@ -0,0 +1,213 @@
+<?php
+
+/** Sanitize the input of a form. Preferable to do on all fields because of form/request forging */
+function clean($data)
+{
+    $data = trim($data);
```
Comment (Drestin) [inline utils/form_validation.php:6]: I have a deprecation warning when `$data == null`. It happens when I open the page "create an event" => I think the error occurs at form init.

File: pages/events/event_edit.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            "end_date" => $event["arrivee"],
+            "limit_date" => $event["limite"]
+        ];
+    }
+}
+
+$v = validate($post);
+$event_name = $v->string("event_name")->label("Nom de l'événement")->placeholder()->required();
+$start_date = $v->date("start_date")->label("Date de départ")->required();
+$end_date = $v->date("end_date")->label("Date de retour")->required()->after($start_date->value);
```
Comment (Drestin) [inline pages/events/event_edit.php:23]: Maybe we should forbid the creation of events if any of the dates is before the request date?

File: utils/form_validation.php [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+    function render(string $attrs = "")
+    {
+        return parent::render("type = date");
+    }
+
+    /** Set upper date limit
+     * @todo better naming?
+     */
+    function before(string $date, string $msg = null)
```
Comment (Drestin) [inline utils/form_validation.php:243]: When I leave `$start_date` empty, there are two errors : "required" and "too soon".
I would prefer if there wasn't the "too soon" error. 
Maybe skip the checks (date, string length, int comparisons ...) if the field was not filled?

**Review by Drestin — CHANGES_REQUESTED** [review]

File: utils/form_validation.php (hunk 3 above)
Comment (Atmos4) [reply utils/form_validation.php:243]: I think I had a previous iteration without stacking checks. I will revert to that :)

File: pages/events/event_edit.php (hunk 2 above)
Comment (Atmos4) [reply pages/events/event_edit.php:23]: Good point. I will also fix being allowed to set the limit date the same day as the event itself 🙈

File: utils/form_validation.php (hunk 1 above)
Comment (Atmos4) [reply utils/form_validation.php:6]: Mmm that's weird. I guess I could add a null check then 🙂

File: pages/events/event_edit.php (hunk 2 above)
Comment (Atmos4) [reply pages/events/event_edit.php]: Done!

File: utils/form_validation.php (hunk 1 above)
Comment (Atmos4) [reply utils/form_validation.php]: Done!

File: utils/form_validation.php (hunk 3 above)
Comment (Atmos4) [reply utils/form_validation.php:241]: Done!

**Review by Mathieu414 — APPROVED** [review]
LGTM

**Pull request merged**